src/pathOrderOptimizer.cpp

```diff
@@ -126,13 +92,78 @@ int PathOrderOptimizer::getClosestPointInPolygon(Point prev_point, int poly_idx)
     {
         const Point& p1 = poly[point_idx];
         const Point& p2 = poly[(point_idx + 1) % poly.size()];
-        int64_t dist = vSize2(p1 - prev_point);
-        float is_on_inside_corner_score = -LinearAlg2D::getAngleLeft(p0, p1, p2) / M_PI * 5000 * 5000; // prefer inside corners
-        // this score is in the order of 5 mm
-        if (dist + is_on_inside_corner_score < best_point_score)
+        // when type is SHARPEST_CORNER, actual distance is ignored, we use a fixed distance and decision is based on curvature only
+        int64_t dist = (type == EZSeamType::SHARPEST_CORNER)? 100 : vSize2(p1 - prev_point);
+        const float corner_angle = LinearAlg2D::getAngleLeft(p0, p1, p2) / M_PI; // 0 -> 2
+        const EZSeamCornerPrefType corner_pref = (config)? config->corner_pref : EZSeamCornerPrefType::Z_SEAM_CORNER_PREF_NONE;
+        if (type == EZSeamType::SHORTEST)
```

Comment:
There's a lot of code duplication between this if branch and the else branch.
You can simplify it a lot by combining the two branches and merging the corner_shift with the dist/scaling_factor:
```
const int64_t corner_shift = (type == EZSeamType::SHORTEST)? 10000 * 10000 : dist / 50;
```